app/scripts/controllers/swaps.js

```diff
@@ -455,22 +485,50 @@ export default class SwapsController {
               ),
               10,
             )
-            .minus(tokenConversionRate ? ethFee.toString(10) : 0, 10)
+            .minus(tokenConversionRate ? totalEthCost : 0, 10)
+
+      sumOfAllTradeValues = sumOfAllTradeValues.plus(ethValueOfTrade)
+      sumOfAllEthFees = sumOfAllEthFees.plus(ethFee)
 
       if (
         ethValueOfTradeForBestQuote === null ||
         ethValueOfTrade.gt(ethValueOfTradeForBestQuote)
       ) {
         topAggId = aggregator
         ethValueOfTradeForBestQuote = ethValueOfTrade
+        ethFeeForBestQuote = ethFee
       }
     })
 
     const isBest =
       quotes[topAggId]?.destinationTokenInfo?.symbol === 'ETH' ||
       Boolean(tokenConversionRates[quotes[topAggId]?.destinationToken])
 
-    return { topAggId, isBest }
+    let savings = null
+
+    if (isBest) {
+      savings = {}
+      // Performance and fee savings are calculated as:
+      //   valueForBestTrade - averageValueOfAllTrades
+      // Where the average is the statistical mean:
+      //   sumOfAllValues / numberOfValues
+
+      savings.performance = ethValueOfTradeForBestQuote.minus(
+        sumOfAllTradeValues.dividedBy(numQuotes),
+        10,
+      )
+
+      savings.fees = ethFeeForBestQuote.minus(
+        sumOfAllEthFees.dividedBy(numQuotes),
```

Comment:
This should actually be:
```
savings.fees = sumOfAllEthFees.dividedBy(numQuotes).minus(
    ethFeeForBestQuote,
    10,
)
```

Reply:
If I would have spent $10 on an apple from Alice, and $15 on an apple from Bob, but I ended up buy an apple from Carol for $5, then the average cost of apples is $10 and my savings against the average is $5.

(($10 + $15 + $5) / 3) - $5 = $5

Reply:
This has been resolved!